### ISSUE TITLE
Apply layout to all available keyboards if you omit the input device

### ISSUE THREAD

**Issue by oblitzitate** [kmonad/kmonad#567]
It would be convenient for `kmonad` to scan all available keyboards in `/dev/input` if you don't specify an input device. 

I use a laptop with an external keyboard, but sometimes like to unplug and carry it around. Such a feature would make it convenient so that I don't have to rely on a script to iterate through all keyboard devices.

I believe `xremap` does this pretty well when you omit the `--device` option. It lets you know which keyboards failed to grab and which are selected.

**Comment by david-janssen** [comment]
Have you checked the `/dev/input/by-id` subdirectory? It usually contains a symlink to your keyboards by an unchanging id, so no scanning required.

**Comment by oblitzitate** [comment]
Yes, but since I use multiple keyboards (e.g. laptop keyboard, usb keyboard 1, usb keyboard 2, etc.), I still have to run `kmonad` for each keyboard. Note: laptop keyboards are located in `/dev/input/by-path`.

Obviously, I can easily do this all with a script, but it'd be nice if it was built-in where I'd just call `kmonad` once and don't need to put an input device in `defcfg`.

**Comment by oblitzitate** [comment]
I don't know if this is the ideal way to do it, but this is the script I use:

```bash
#!/usr/bin/env bash
kbd_file="$1"
fd '^(usb|platform).+event-kbd$' /dev/input | while read keyboard; do
  kmonad -w 0 --input "device-file \"$keyboard\"" "$kbd_file" &
done
```

**Comment by maueroats** [comment]
Suggestion: put a link to this information in the tutorial by the `device-file` Linux information. 
Use the link to [obliterate's comment](https://github.com/kmonad/kmonad/issues/567#issuecomment-1154233223). 

The command-line arguments were not very discoverable to me. I think changing the help string for `--input` would help a bit. It you could add "Specifies the `device-file`" to make it a little clearer. Or call the value `DEVICE` or even `INPUT_DEVICE` instead of `ITOKEN`.

Related:

You could consider a section in the FAQ: "What command line arguments are available for kmonad?"

**Comment by david-janssen** [comment]
> Suggestion: put a link to this information in the tutorial by the device-file Linux information.

Pullrequests are welcomed :)

**Comment by tobiasBora** [comment]
If anyone has a script to make it reload when a new keyboard is plugged, please let me know ;-)

**Comment by tsankuanglee** [comment]
> If anyone has a script to make it reload when a new keyboard is plugged, please let me know ;-) I'd love kmonad to deal with hotplug out of the box ^^

This can be done with udev rules. Upon hotplug, let it set `SYSTEMD_WANTS` so a systemd service can be loaded:

This is my working setup (I have a bunch of similar devices so my conditions are very verbose):
```
# /etc/udev/rules.d/54-remote_control.rules
SUBSYSTEM=="input", KERNEL=="event*", ATTRS{id/bustype}=="0003", ATTRS{id/product}=="8367", ATTRS{id/vendor}=="248a", ATTRS{id/version}=="0110", ATTRS{name}=="XING WEI 2.4G USB USB Composite Device Consumer Control", SYMLINK+="input/by-id/XING_WEI_MX3_REMOTE-2.4G-kbd-multimedia", TAG+="systemd", ENV{SYSTEMD_WANTS}="kmonad@MX3-multimedia.service"
```

The service file will call kmonad with your specified config, which can also be parameterized.
```
# /etc/systemd/system/kmonad@.service
[Unit]
Description=kmonad for %i
After=network.target
StartLimitIntervalSec=0

[Service]
ExecStart=/usr/bin/kmonad /etc/kmonad/remote-%i.kbd
Restart=no
RestartSec=1

[Install]
WantedBy=multi-user.target
```

I'm hard-coding my device node since it's very specific for my use case, but you can always change it to whatever your udev rule sets.
```
;; part of defcfg
  input  (device-file "/dev/input/by-id/XING_WEI_MX3_REMOTE-2.4G-kbd-multimedia")
```

**Comment by tsankuanglee** [comment]
> > Suggestion: put a link to this information in the tutorial by the device-file Linux information.
> 
> Pullrequests are welcomed :)

https://github.com/kmonad/kmonad/pull/993
